### ISSUE TITLE
[General Responsiveness]: some clicks are eaten=not executed

### ISSUE THREAD

**Issue by star-buck** [blue-systems/plasma-5.4#29]
1. MENU-SWITCH:
   switch to alternative menu (kickoff to kicker or viceversa)
   first click on startbutton doesnt do anything (kubuntu-plasma5-201412051356-amd64.iso )
2. TASKS:
   right click on task-manager in panel area to bring up dialog, click on desktop to close it:
   next mouse click eaten, like clicking on menu button does nothing.
3. ICON in panel:
   right click on any icon in panel
   left-click somewhere on the desktop to close popup from right-click
   click on "Panel Edit" mode icon does nothing, only second click activates "edit mode"
4. WINDOW DRAG:
   click on top window bar to focus window: you can right-click on desktop
   click and hold to drag window to move around: you can not right-click on desktop after that (was possible under kde4.x)
5. KMIX (FIXED WITH NEW APPLET):
   left-click on kmix icon in systray
   while kmix dialog is open, try to grab and move a window from window titlebar

**Comment by star-buck** [comment]
tested with 20150109:
right click on task-manager in panel area to bring up dialog, click on desktop to close it:
next mouse click eaten, like clicking on menu button does nothing.

IMO this is super-annoying and will certainly be noticed by anyone using plasma for more than a day and seen then as a sign of poor quality.

**Comment by mgraesslin** [comment]
can you test some more things: please try clicking on the task manager to close it. For me, the next event is not eaten in that case. If you can confirm that I have a theory to create a bug report against Qt.

**Comment by star-buck** [comment]
confirmed

**Comment by star-buck** [comment]
maybe i found some major clue:
- change to kicker
- right-click on an icon from menu category, for example dolphin:
- select "Edit application" to bring up popup to edit
- while popup is open, ALL mouseclicks are eaten :)

**Comment by eikehein** [comment]
The last thing is normal actually, it's a modal dialog and plasmashell is the parent app.

**Comment by star-buck** [comment]
makes sense then :)
still some feeling this could be actually have something to do with the whole parent app/window, panel being also a window, etc....

**Comment by mgraesslin** [comment]
I am quite sure to have understood the root cause of the problem
1. MouseArea gets a mouse press event
2. The mouse press event triggers a context menu
3. The context menu performs grabKeyboard
4. The context menu is closed by clicking somewhere else, keyboard is released
5. The MouseArea didn't know anything about the keyboard grab and still thinks it is in pressed state
6. Next mouse press event on MouseArea is ignored as it's in wrong state

I'm sure to be able to construct a minimal example to illustrate the problem to Qt devs, but I'm not sure how and where this could be fixed.

**Comment by notmart** [comment]
Reported as upstream bug
https://bugreports.qt.io/browse/QTBUG-43776

**Comment by star-buck** [comment]
any progress so it can be hotfixed/patched?

**Comment by eikehein** [comment]
Not AFAIK. I think we need to try to make some waves about this and annoy some people on IRC about it to get more eyes on the bug. Trying to find drgvond on #qt-labs or maybe talking to Thiago or fregl or kfunk might be a good idea.

**Comment by notmart** [comment]
I'm not sure is 100% correct (and if will be accepted) but a tentative Qt patch is here:
https://codereview.qt-project.org/#/c/104061/

**Comment by star-buck** [comment]
@notmart : what package would this land in?
can we hotfix that package for kci?

**Comment by davidedmundson** [comment]
Qt.

Currently we don't build our own version of that.

**Comment by star-buck** [comment]
But which package, so we can patch it?

**Comment by mgraesslin** [comment]
The package is libqt5gui5. Please note that the patch has regression potential.

**Comment by notmart** [comment]
A ppa could provide a patched qtgui5 that would override the system one.

**Comment by star-buck** [comment]
i noticed this:
- click on top window bar to focus window: you can right-click on desktop
- click and hold to drag window to move around: you can not right-click on desktop after that (was possible under kde4.x)

**Comment by apachelogger** [comment]
https://launchpad.net/~netrunner-os/+archive/ubuntu/netrunner-15-ci-fixes/+sourcepub/4691840/+listing-archive-extra

**Comment by star-buck** [comment]
still fails here even with updated fixes packages.

**Comment by notmart** [comment]
trying  on the netrunner live image, did the following
open taskbar context menu
click on another task dismisses the menu
another click without moving the mouse activates the task

this is the correct behavior, any particular sequence of steps to reproduce the problem?

**Comment by star-buck** [comment]
Switch menus via alternatives
Click on startbutton not executed

**Comment by notmart** [comment]
right after switching?

**Comment by star-buck** [comment]
Of course right after switching!
People switch menus and then click on the startbutton right after switching
estimated 99,99999999% to see if its working.
Or are you saying the first click on the menu right after switching is
executed at your computer?
Am 10.02.2015 13:31 schrieb "Marco Martin" <notifications@github.com>:

**Comment by eikehein** [comment]
The good news is that it does work fine here (Qt 5.4 branch git which should have Marco's patch, Plasma git master which probably doesn't matter), so it has to be possible to get it working in principle ...

**Comment by notmart** [comment]
in the case of applet switching seems to be reproducible locally too (oddly, some times, doesn't happen 100% of the times).
I'll look into it, seems to be some other kind of issue comared to the original  popup dismissing one
@eikehein: so does it work for you after the switch?

**Comment by notmart** [comment]
doesn't seem directly related to the switching, even just opening the alternatives menu eats a click.
if you unblock it before performing the switch, then you switch and the first click will be ok.
so the problem is in the chain context menu->alternatives menu

**Comment by eikehein** [comment]
Here is my steps:

1. Right-click Kicker (or Kickoff)
2. Click Alternatives
3. Switch to the other menu
4. Click the menu icon

-> Menu opens.

**Comment by notmart** [comment]
interesting, wonder if is a different qt revision

**Comment by star-buck** [comment]
this is also not fixed: https://github.com/blue-systems/plasma-5.2/issues/29#issuecomment-65898022

**Comment by notmart** [comment]
this commit
http://quickgit.kde.org/?p=plasma-workspace.git&a=commit&h=71b35c8b06220c1c31d29665c0559feb11055687
seem to fix the click issue with the alternatives dialog

**Comment by star-buck** [comment]
maybe it didnt got shipped, but here the annoying issue can still be seen in action:
https://www.youtube.com/watch?v=Dh7A7L-3KcQ#t=168

**Comment by notmart** [comment]
the commit is in master and 5.2, locally i still indeed see the problem happen sometimes

**Comment by star-buck** [comment]
so the bug is still not fixed, as it happens here all the time, not sometimes...

**Comment by star-buck** [comment]
lets check again on fridays kci and prepare to fix this once more.

**Comment by notmart** [comment]
So, the problem seems is related but not the same fixed by the Qt patch.
I pushed in plasma-framework a patch to this problem in particular and is in this commit:
http://commits.kde.org/plasma-framework/1794d3c43f9a1b60b1fd24946a865ba43352f50e

so it needs either an updated plasma-framework package or the above patch to be included.
